cmd/cli/operations/retrieve_savepoint.go

```diff
@@ -2,12 +2,78 @@ package operations
 
 import (
 	"errors"
+	"net/url"
+	"os"
 	"strings"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/spf13/afero"
 )
 
+var s3Schemes = map[string]bool{
+	"s3":  true,
+	"s3a": true,
+	"s3p": true,
+}
+
 func (o RealOperator) retrieveLatestSavepoint(dir string) (string, error) {
+	u, err := url.Parse(dir)
+
+	if err == nil && s3Schemes[u.Scheme] {
+		return o.retrieveLatestSavepointS3(u)
```

Comment:
There could be multiple jobs running in the same flink cluster. Do you assume the savepoint dir param to this function is unique to each job?